api.go

```diff
@@ -228,9 +228,15 @@ func RecoverCluster(conf *Config, fsm FSM, logs LogStore, stable StableStore,
 		return err
 	}
 
-	// Sanity check the Raft peer configuration.
-	if err := checkConfiguration(configuration); err != nil {
-		return err
+	// Sanity check the Raft peer configuration. Note that it's ok to
+	// recover to an empty configuration if you want to keep the data
+	// but not allow a given server to participate any more (you'd have
+	// to recover again, or add it back into the existing quorum from
+	// another server to use this server).
```

Comment:
I don't think that's safe to do. It'd be marking entries committed that may not have been, and the other servers may still think this one is part of the cluster.

In general, it's never safe to call RecoverCluster. When users invoke it manually, they should know that they're doing something risky and have signed the appropriate waivers. It's not cool to invoke automatically. It's not a hammer you can just pick up and use that way.